File: CarlaErik/UvA.SoftCon.Questionnaire/AST/Expressions/BinaryExpression.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace UvA.SoftCon.Questionnaire.AST.Expressions
{
    public abstract class BinaryExpression : IExpression
    {
        public IExpression Left
        {
            get;
            private set;
        }

        public IExpression Right
        {
            get;
            private set;
        }

        protected BinaryExpression(IExpression left, IExpression right)
        {
            Left = left;
            Right = right;
        }
    }
}

File: Stravopodis_Safaric/QL/src/uva/ql/ast/expressions/BinaryExpression.java
package uva.ql.ast.expressions;
import uva.ql.ast.CodeLines;

public abstract class BinaryExpression extends Expression{

	private Expression left;
	private Expression right;
	private Operator operator;
	
	public BinaryExpression(Expression _left, Expression _right, Operator _operator, CodeLines _codeLines) {
		super(_codeLines);
		
		this.left = _left;
		this.right = _right;
		this.operator = _operator;
	}
	
	public Expression getLeftExpr(){
		return this.left;
	}
	
	public Expression getRightExpr(){
		return this.right;
	}
	
	public Object getLeftExpressionValue(){
		return this.left.getValue();
	}
	
	public Object getRightExpressionValue(){
		return this.right.getValue();
	}
	
	public Operator getOperator(){
		return this.operator;
	}
}
